tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,348 @@
+"""Test the BryantEvolutionClient type."""
+
+from contextlib import contextmanager
+from datetime import timedelta
+import itertools
+import logging
+from unittest.mock import AsyncMock, patch
+
+from evolutionhttp import BryantEvolutionLocalClient
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from homeassistant.components.bryant_evolution.climate import SCAN_INTERVAL
+from homeassistant.components.bryant_evolution.const import CONF_SYSTEM_ZONE, DOMAIN
+from homeassistant.components.climate import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODE,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACAction,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ENTITY_ID, CONF_FILENAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util.unit_system import US_CUSTOMARY_SYSTEM
+
+from .conftest import DEFAULT_SYSTEM_ZONES
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@contextmanager
+def disable_auto_entity_update():
+    """Context manager to disable auto entity updates."""
+    with patch(
+        "homeassistant.helpers.entity.Entity.async_update_ha_state", return_value=None
+    ) as patch_update:
+        yield patch_update
```

Comment:
Why do we do this exactly?

Reply:
This is part of my quest to verify that we call async_update_ha_state in the climate services that modify the climate entity state, such as set_fan_speed. Recall that we want to make this call because polling this device is very slow, and we want to ensure that changes the user has made are reflected in the UI more quickly than they would be if we waited 10+ seconds for a polling event to complete.

After invoking the entity service handler, HA will trigger async_update for entities that use polling [1]. This happens synchronously, before the service call completes from the point of view of the caller.

Because of this auto-poll, the externally-visible effects of the service call are identical whether or not we cal async_update_ha_state, which makes it hard to write a unit test. 

This function patches the HA core to make the auto-poll a no-op. This way, if we see that the entity was updated, we know it must have been through async_update_ha_state.

Three other possible approaches:

1) Patch async_update_ha_state to be a mock and verify that it was called. IMHO this is a weaker test, because we wouldn't be verifying the end-to-end behavior (we'd just be checking we made a function call, not that the state is visibile end-to-end).
2) Modify HA production code to allow disabling auto-poll-after-service-call for entities that don't need it.
3) Change the Bryant Evolution climate platform not to be a polling-type entitity (to avoid the auto-poll-after-service-call). Manually schedule polling. 


1. https://github.com/home-assistant/core/blob/162b734be756d48e5e36ec828e6703110b7e2f59/homeassistant/helpers/service.py#L1006

Reply:
changed to use mocks as discussed